catroid/src/main/java/org/catrobat/catroid/ui/fragment/ProjectListFragment.java

```diff
@@ -369,24 +372,28 @@ public boolean onOptionsItemSelected(MenuItem item) {
 	}
 
 	private void initializeDefaultProjectAfterDelete() {
-		final ProjectManager projectManager = ProjectManager.getInstance();
-		getActivity().findViewById(R.id.fragment_container).setVisibility(View.GONE);
-		getActivity().findViewById(R.id.progress_circle).setVisibility(View.VISIBLE);
-		Runnable r = new Runnable() {
+
+		projectInitializer = new AsyncTask<Void, Void, Activity>() {
+
 			@Override
-			public void run() {
-				projectManager.initializeDefaultProject(getActivity());
-				getActivity().runOnUiThread(new Runnable() {
-					@Override
-					public void run() {
-						getActivity().findViewById(R.id.fragment_container).setVisibility(View.VISIBLE);
-						getActivity().findViewById(R.id.progress_circle).setVisibility(View.GONE);
-						initializeList();
-					}
-				});
+			protected void onPreExecute() {
+				getActivity().findViewById(R.id.fragment_container).setVisibility(View.GONE);
+				getActivity().findViewById(R.id.progress_circle).setVisibility(View.VISIBLE);
 			}
-		};
-		(new Thread(r)).start();
+
+			@Override
+			protected Activity doInBackground(Void... params) {
+				ProjectManager.getInstance().initializeDefaultProject(getActivity());
+				return getActivity();
+			}
+
+			@Override
+			protected void onPostExecute(Activity activity) {
+				activity.findViewById(R.id.fragment_container).setVisibility(View.VISIBLE);
+				activity.findViewById(R.id.progress_circle).setVisibility(View.GONE);
+				initializeList();
+			}
+		}.execute();
```

Comment:
I took a look at this. IMO the best way to handle it is using a variable for the Activity.
```java
final Activity activity = getActivity();
projectInitializer = new AsyncTask<Void, Void, Void>() {
…
```
In `onPreExecute` and `onPostExecute` there is no null check needed because these methods run on the ui thread.
`doInBackground` is only used as context.

However and JFI, using `getActivity()` in `doInBackground` may result in a NPE.

Reply:
So, should I use a variable "activity" and apply the null checker in "doInBackground'?

Reply:
variable - yes, null check - no.

The benefit of using a variable here is that `initializeDefaultProject` will only use the activity as context. So passing in a finished activity should be fine.